src/include/security.h

```diff
@@ -89,14 +89,42 @@ pgmoneta_get_master_key(char** masterkey);
 int
 pgmoneta_tls_valid(void);
 
+/**
+ * Common function for generating hash for a file
+ * @param filename The file path
+ * @param hash The hash value
+ * @param md_name The md type(SHA256, SHA384, SHA512)
+ * @return 0 upon success, otherwise 1
+ */
+int
+pgmoneta_create_hash_file(char* filename, char** hash, const char* md);
```

Comment:
I think this should be a `static` function inside `security.c` called `create_hash_file()`.

And, it should be `char* filename, char* algorithm, char** hash)` as the signature, first input then output after